Jenkinsfile

```diff
@@ -12,176 +12,179 @@ repo = tokens[tokens.size()-2]
+                    buildUwpx64Release  : doBuildWindows('Release', true, 'x64', false),
+                    buildUwpArmDebug    : doBuildWindows('Debug', true, 'ARM', false),
+                    buildUwpArmRelease  : doBuildWindows('Release', true, 'ARM', false),
+
+                    buildLinuxDebug     : doBuildLinux('Debug'),
+                    buildLinuxRelease   : doBuildLinux('Release'),
+                    buildLinuxRelAssert : doBuildLinux('RelAssert'),
+                    buildLinuxASAN      : doBuildLinuxASAN()
+                ]
+
+                androidAbis = ['armeabi-v7a', 'x86', 'mips', 'x86_64', 'arm64-v8a']
+                androidBuildTypes = ['Debug', 'Release']
+
+                for (def i = 0; i < androidAbis.size(); i++) {
+                    def abi = androidAbis[i]
+                    for (def j = 0; j < androidBuildTypes.size(); j++) {
+                        def buildType = androidBuildTypes[j]
+                        parallelExecutors["android-${abi}-${buildType}"] = doAndroidBuildInDocker(abi, buildType, false)
+                    }
+                }
 
-      stage('Aggregate') {
-          parallel (
-            cocoa: {
-                  node('docker') {
-                      getArchive()
-                      for (int i = 0; i < cocoaStashes.size(); i++) {
-                          unstash name:cocoaStashes[i]
-                      }
-                      sh 'tools/build-cocoa.sh'
-                      archiveArtifacts('realm-core-cocoa*.tar.xz')
-                      if(gitTag) {
-                          def stashName = 'cocoa'
-                          stash includes: 'realm-core-cocoa*.tar.xz', name: stashName
-                          publishingStashes << stashName
-                      }
-                  }
-              },
-            android: {
-                  node('docker') {
-                      getArchive()
-                      for (int i = 0; i < androidStashes.size(); i++) {
-                          unstash name:androidStashes[i]
-                      }
-                      sh 'tools/build-android.sh'
-                      archiveArtifacts('realm-core-android*.tar.gz')
-                      if(gitTag) {
-                          def stashName = 'android'
-                          stash includes: 'realm-core-android*.tar.gz', name: stashName
-                          publishingStashes << stashName
-                      }
-                  }
-              }
-          )
-      }
+                appleSdks = ['ios', 'tvos', 'watchos']
+                appleBuildTypes = ['MinSizeDebug', 'Release']
 
-      if (gitTag) {
-          stage('publish-packages') {
-              parallel(
-                  generic: doPublishGeneric(),
-                  others: doPublishLocalArtifacts()
-              )
-          }
-      }
-  }
-}
+                for (def i = 0; i < appleSdks.size(); i++) {
```

Comment:
Prettier to do `appleSdks.each() { sdk -> ... }` (and similar in the loop below)